docs/features/software-templates/migrating-from-v1alpha-to-v1beta2.md

```diff
@@ -0,0 +1,328 @@
+---
+id: migrating-from-v1alpha-to-v1beta2
+title: Migrating to v1beta2 templates
+description:
+  Help on how to move your old templates from v1alpha1 to the more declarative
+  v1beta2
+---
+
+# What's new?
+
+Previously, the scaffolder was very restricted in what you could when creating
+new software components from templates. There were three steps of scaffolding
+which was pretty hard to extend and add new functionality to, and it was pretty
+hard to re-use logic between templates. The old way of having `preparers`
+`templaters` and `publishers` which was defined by the backend and needed to be
+run for each template has changed to give the template total control over what
+should be run part of the templating run. This makes templates a little more
+declarative as you can now register different `actions` or `functions` with the
+`scaffolder-backend` which you then can decide how you want to run them and in
+what order in the template definition `yaml` file.
+
+We've also made some improvements and added some helpers to work with
+cookiecutter and storing the skeleton for a template in a different place to
+where you're entity definition is, previously you needed to have your
+`template.yaml` next to the skeleton source (`{{cookiecutter.component_id}}`
+directory), but now that's not the case. Part of the changes with the `v1beta2`
+syntax is that you can grab your template source from any repository, and re-use
+them between templates.
+
+We've also renamed the `schema` property to `parameters` as this makes more
+sense when using them as parameters to the actions or steps that you've setup
+for your templates. There's now the added benefit that you can now assign an
+array to the `parameters` property, which will then give you multiple steps in
+the UI, so you can split apart your input parameter and group them as needed
+rather than having one long list of input fields.
+
+## The `parameters` property
```

Comment:
this section seems to be mostly a rewording of the above - should we skip the one above or move some of it into this section?

Reply:
It's more of a technical explanation about what is happening with the change. Rather than just the benefits. I think it makes sense to keep both.

Reply:
Aight!